flask/src/main.py

```diff
@@ -473,6 +474,57 @@ def showSuggestion():
 
     return jsonify({"response": has_openai_key}), 200
 
+@app.route('/apply-promo-code', methods=['POST'])
+def apply_promo_code():
+    logger.info('[/apply-promo-code] request received')
+
+    try:
+        body = json.loads(request.data)
+        promo_code = body.get('value', '').strip()
+        
+        if not promo_code:
+            logger.warning('[/apply-promo-code] bad request - missing value parameter')
+            return '', 400
+        
+        promo_code_data = get_promo_code(promo_code)
+        
+        if not promo_code_data:
+            logger.warning('[/apply-promo-code] code not found: %s', promo_code)
+            return jsonify({
+                "error": {
+                    "code": "not_found",
+                    "message": "Promo code not found."
+                }
+            }), 404
+        
+        promo_dict = dict(promo_code_data)
+        logger.info('[/apply-promo-code] code found: %s', promo_dict)
+        
+        if promo_dict.get('expires_at') and promo_dict['expires_at'] <= datetime.now():
```

Comment:
I didn't dig into this but just running through the flow (https://kosty-react-dot-sales-engineering-sf.appspot.com/) i got a 404. Should i have gotten a 410?

<img width="1909" height="1124" alt="image" src="https://github.com/user-attachments/assets/3013f940-466f-47ef-82bb-2cd9bcd763d3" />

Reply:
yup, works fine in staging-react:
<img width="1235" height="83" alt="Screenshot 2025-09-19 at 2 56 10 PM" src="https://github.com/user-attachments/assets/31923a5c-1a3d-4515-a465-36274f03c9fd" />